aopy/analysis/base.py

```diff
@@ -366,10 +366,34 @@ def calc_freq_domain_amplitude(data, samplerate, rms=False):
     freq_data = np.fft.fft(data, axis=0)
     length = np.shape(freq_data)[0]
     freq = np.fft.fftfreq(length, d=1./samplerate)
-    data_ampl = abs(freq_data[freq>=0,:])*2/length # compute the one-sided amplitude
+
+    # Only take non-negative frequency components
     non_negative_freq = freq[freq>=0]
+    non_negative_freq_data = freq_data[freq>=0,:]/complex(length,0) # normalize by length
+    non_negative_freq_data[1:,:] = non_negative_freq_data[1:,:]*2 # account for half the peak amplitude being at the negative frequency component
```

Comment:
to be clear, it's [1: ] bcz fft gives you dc bias for it's first variable?

Reply:
also confused what happens if freqs is an odd number? or is it never an odd number?

Reply:
Yes, because the 0th index is the dc offset

Reply:
Not sure I understand what you mean by if freqs is an odd number?

Reply:
oh if the number of frequencies is odd

Reply:
I don't think it matters. The function will return all non-negative freqs and their corresponding freq-domain values, including the dc offset, regardless of # of freqs. I'm multiplying all the values at positive frequencies by 2 to account for there being a symmetrical value at the negative frequency component. The freq-domain values are symmetrical around freq=0, so the dc offset value doesn't need to be corrected. Does this make sense?